src/celldega/pre/image_quantification.py

```diff
@@ -0,0 +1,165 @@
+import os
+import glob
+import tifffile
+import pandas as pd
+import numpy as np
+import geopandas as gpd
+import matplotlib.pyplot as plt
+from rasterio.mask import mask
+from rasterstats import zonal_stats
+from shapely.geometry import Polygon, MultiPolygon
+from shapely.affinity import scale
+from PIL import Image, ImageEnhance
+from ..pre.boundary_tile import numpy_affine_transform, batch_transform_geometries
+
+def process_row(row):
+    geometry = row['Geometry']
+    if geometry.geom_type == "MultiPolygon":
+        largest_polygon = max(geometry.geoms, key=lambda p: p.area)
+        row['Geometry'] = largest_polygon
+    return row
+
+def image_contrast_adjustment(image_file, base_path, contrast_factor=5, intensity_threshold=100):
+    stain_image = Image.open(image_file).convert("L")
+
+    # Plot the original stain image
+    plt.figure(figsize=(10, 10))
+    plt.imshow(stain_image, cmap='gray')
+    plt.title("Original Stain Image")
+    plt.show()
+
+    enhancer = ImageEnhance.Contrast(stain_image)
+    stain_image_with_contrast = enhancer.enhance(contrast_factor)
+
+    # Plot the stain image after contrast adjustment
+    plt.figure(figsize=(10, 10))
+    plt.imshow(stain_image_with_contrast, cmap='gray')
+    plt.title(f"Stain Image with Contrast Factor {contrast_factor}")
+    plt.show()
```

Comment:
I saw a few `plt.show()` in this file. If it is standard QC results plot, it is better to write to a file. If this function is called multiple time, you may want to consider skipping the `plt.show()` with `plt.close() or add a bool argument to switch on/off displaying plot.